File: scripts/check_repo.py
import subprocess
import sys
from pathlib import Path


ROOT = Path(__file__).resolve().parents[1]


def run_node_script(relative):
    return subprocess.run(
        ["node", str(ROOT / relative)],
        cwd=ROOT,
        text=True,
        capture_output=True,
        check=False
    )


def main():
    errors = []

    for script in ["scripts/check_repo_structure.js", "scripts/check_viewer_geometry.js"]:
        result = run_node_script(script)
        if result.returncode:
            errors.append(f"{script} failed")
            for line in (result.stdout + result.stderr).splitlines():
                errors.append(f"{script}: {line}")

    if errors:
        print("FAILED: repository checks failed")
        for error in errors:
            print(f"ERROR: {error}")
        return 1

    print("OK: repository checks passed")
    return 0


if __name__ == "__main__":
    sys.exit(main())

File: scripts/check_repo.js
const path = require("path");
const { spawnSync } = require("child_process");

const ROOT = path.resolve(__dirname, "..");
const CHECKS = [
  "scripts/check_branch_guard.js",
  "scripts/check_repo_structure.js",
  "scripts/check_reference_geometry_translator.js",
  "scripts/validate_domain_model.js",
  "scripts/check_viewer_runtime.js"
];

for (const script of CHECKS) {
  const result = spawnSync(process.execPath, [path.join(ROOT, script)], {
    cwd: ROOT,
    stdio: "inherit"
  });
  if (result.status !== 0) process.exit(result.status || 1);
}
